pkg/ruler/api.go

```diff
@@ -321,7 +321,12 @@ var (
 	ErrBadRuleGroup = errors.New("unable to decode rule group")
 )
 
-func marshalAndSend(output interface{}, w http.ResponseWriter, logger log.Logger) {
+type HTTPHeader struct {
```

Comment:
[non blocking] I don't see the need to define our own data structure when we could just `http.Header` (`http.Header` is a map of headers, but it works fine for your purpose).

Reply:
I tried to that originally, but now I can't remember why I abandoned that idea - let me give it another try now that the feature is complete.

Reply:
WDYT? https://github.com/grafana/mimir/pull/8444/commits/ece4cf736430b547fc803e272d92501547a4ce2d